Essentials/src/com/earth2me/essentials/utils/FormatUtil.java

```diff
@@ -38,6 +43,30 @@ public static String stripEssentialsFormat(final String input) {
         return stripColor(input, REPLACE_ALL_PATTERN);
     }
 
+    public static String placeholderAPIFormat(final IUser user, final String input) {
+        return placeholderAPIFormat(user.getBase(), input);
+    }
+
+    //Formatting PlaceholderAPI Placeholders, returns the original string if the plugin isn't enabled
+    //Seperate this from #getChatFormat() in ISettings due to the requirement of Player as an argument
+    public static String placeholderAPIFormat(final Player player, String input) {
+        if (input == null) {
+            return null;
+        }
+
+        input = input.replaceAll("\\{|\\}", "%");
```

Comment:
There might be some problem that a single { standing alone will become % after that. I don't use regex for a long time. Should we do something about it.

Reply:
`/{([A-Za-z0-9_]+)}/g` might work.

Reply:
@md678685 I love IntelliJ
![image](https://user-images.githubusercontent.com/10562127/38250075-c80b862e-3788-11e8-8755-ee74b575364c.png)

Reply:
Still need to address this.

Reply:
I think the replacement with that regex would have been `"%$1%"`, since the `()`s in the regex capture the placeholder name (`bobsplugin_hunger`) and the `$1` puts it back in the string once replaced.